client/src/game/api/events/shape/core.ts

```diff
@@ -29,8 +29,15 @@ socket.on("Shape.Set", (data: ApiShapeWithLayerInfo) => {
     const old = getShapeFromGlobal(uuid);
     const isActive = activeShapeStore.state.id === getLocalId(uuid);
     const hasEditDialogOpen = isActive && activeShapeStore.state.showEditDialog;
-    if (old) old.layer?.removeShape(old, { sync: SyncMode.NO_SYNC, recalculate: true, dropShapeId: true });
-    const shape = addShape(apiShape, floor, layer, SyncMode.NO_SYNC);
+    let deps = undefined;
+    if (old) {
+        deps = old.dependentShapes;
+        for (const dep of deps) {
+            old?.removeDependentShape(dep.id, { dropShapeId: false });
+        }
```

Comment:
I believe this can be simplified to the following ?

```suggestion
        old.removeDependentShapes({ dropShapeId: false });
```

If not, the `removeDependentShapes` function is always called with `dropShapeId` true, in which case the argument can just be removed alltogether.

Reply:
Ah my bad. I didn't originally have the `removeDependentShapes` function just `removeDependentShape` and decided to add the former for things like this, but forgot to change this old code.